pkg/controller/vpcpeering/peering.go

```diff
@@ -528,12 +528,33 @@ func (e *external) Delete(ctx context.Context, mg cpresource.Managed) error { //
 		}
 	}
 
-	_, err = e.client.DeleteVpcPeeringConnectionRequest(&ec2.DeleteVpcPeeringConnectionInput{
-		VpcPeeringConnectionId: cr.Status.AtProvider.VPCPeeringConnectionID,
-	}).Send(ctx)
+	peeringFilter := ec2.Filter{
+		Name: aws.String("tag:Name"),
+		Values: []string{
+			cr.Name,
+		},
+	}
+
+	describeVpcPeeringConnectionsInput := &ec2.DescribeVpcPeeringConnectionsInput{
```

Comment:
we may keep the filter consistent with the filter used in `Observe()`

Reply:
done.